flasksrc/quiz.py

```diff
@@ -58,3 +85,104 @@ def get(self, quiz_id):
                 )
 
         return {'items': result}, 200
+
+
+class ToyNetQuizScore(MethodResource):
+    @jwt_required()
+    @use_kwargs(ToyNetQuizScorePostReq)
+    @marshal_with(ToyNetQuizScorePostResp)
+    def post(self, **kwargs):
+        try:
+            req = ToyNetQuizScorePostReq().load(kwargs)
+        except ValidationError as e:
+            abort(400, message=f'malformed create submission request: {e.messages}')
+
+        quiz_id = req['quiz_id']
+        user_id = req['user_id']
+        count_correct = req['count_correct']
+        count_wrong = req['count_wrong']
+
+        db = get_db()
+        try:
+            db.execute(
+                'INSERT INTO toynet_quiz_scores(quiz_id,user_id,count_correct,count_wrong)'
+                ' VALUES((?), (?), (?), (?))',
+                (quiz_id, user_id, count_correct, count_wrong,)
+            )
+            db.commit()
+            rows = db.execute(
+                'SELECT scores.submission_id'
+                ' FROM toynet_quiz_scores AS scores'
+                ' WHERE scores.quiz_id = (?) AND scores.user_id = (?)',
+                (quiz_id, user_id,)
+            ).fetchall()
+        except Exception as e:
+            print(e.args[0])
+            abort(400, message="Insertion failed")
+
+        return {'submission_id': rows[-1]['submission_id']}, 201
+
+
+class ToyNetQuizScoresByUser(MethodResource):
+    @jwt_required()
+    @marshal_with(ToyNetQuizScoresGetResp)
+    def get(self, user_id):
+        db = get_db()
+        try:
+            rows = db.execute(
+                'SELECT * FROM toynet_quiz_scores as scores'
+                ' WHERE scores.user_id = (?)'
+                ' GROUP BY scores.quiz_id',
+                (user_id,)
+            ).fetchall()
+        except Exception as e:
+            print(e.args[0])
+            abort(500, message="query for quiz scores failed: {}".format(user_id))
+
+        if not len(rows):
+            return {'scores': []}, 200
+        else:
+            scores = [
+                {
+                    'quiz_id': rows[0]['quiz_id'],
+                    'count_total': rows[0]['count_correct']+rows[0]['count_wrong'],
+                    'scores': [
+                        {
+                            'count_correct': rows[0]['count_correct'],
+                            'datetime': rows[0]['submitted']
+                        },
+                    ]
+                },
+            ]
+
+        curr_quiz = rows[0]['quiz_id']
+        count = 0
```

Comment:
instead of tracking count, you can build a temporary score object and append it when you're done:

say `input_list` is
```
[
    { 'id': 'x', 'time':  3pm, 'value': 'hello'},
    { 'id': 'x', 'time':  5pm, 'value': 'world'}, 
    { 'id': 'y', 'time':  1pm, 'value': 'i'},
    { 'id': 'z', 'time':  9am, 'value': 'am'},
    { 'id': 'z', 'time':  10am, 'value': 'a'},
    { 'id': 'z', 'time':  11am, 'value': 'program'}
]
```

Instead of:
```
curr_id = input_list[0]['id']
count = 0

for input in input_list:
    if input['id'] == curr_id:
        res[count]['values'].append(input['value'])
    else:
        curr_id = input['id']
        count += 1
        res.append({ 'id': input['id'], 'time':  input['time'], 'value': [ input['value'] ] })

return res 
```
you can do:

```
curr_item = { 'id': '' }
count = 0

for input in input_list:
    if curr_item['id'] != input['id']:
        res.append(curr_item)
        curr_item['id'] = { input['id'], 'time':  input['time'], 'value': [ input['value'] ] })
    else:
        curr_item['values'].append(input['value'])

res.append(curr_item) // since last one won't get assigned

return res 
```

Reply:
I tried to follow these suggestions, but I'm seeing the same issue where only the first submission with that quiz_id is added to the object